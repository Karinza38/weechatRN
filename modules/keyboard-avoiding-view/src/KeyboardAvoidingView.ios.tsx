import { requireNativeViewManager } from 'expo-modules-core';
import type * as React from 'react';

import type { KeyboardAvoidingViewProps } from './KeyboardAvoidingView.types';

const NativeView: React.ComponentType<KeyboardAvoidingViewProps> =
  requireNativeViewManager('KeyboardAvoidingView');

export default function KeyboardAvoidingView({
  children,
  style,
  ...props
}: KeyboardAvoidingViewProps) {
  return (
    <NativeView {...props} style={style}>
      {children}
    </NativeView>
  );
}
