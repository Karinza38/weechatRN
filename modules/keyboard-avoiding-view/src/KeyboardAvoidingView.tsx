import { View } from 'react-native';

import type { KeyboardAvoidingViewProps } from './KeyboardAvoidingView.types';

export default function KeyboardAvoidingView({
  children,
  ...props
}: KeyboardAvoidingViewProps) {
  return <View {...props}>{children}</View>;
}
