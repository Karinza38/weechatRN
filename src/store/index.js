import { combineReducers, createStore, applyMiddleware } from 'redux';
import logger from 'redux-diff-logger';

import buffer from '../usecase/buffers/reducers/BufferReducer';


const app = (state = {}, action) => {
    return state;
};

const reducer = combineReducers({
    app,
    buffer
});


const createStoreWithMiddleware = applyMiddleware(logger)(createStore);

export default createStoreWithMiddleware(reducer);