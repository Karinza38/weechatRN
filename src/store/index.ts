import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import {
  FLUSH,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
  REHYDRATE,
  persistReducer,
  persistStore
} from 'redux-persist';

import buffers, { BufferState } from './buffers';
import connection, { ConnectionInfo } from './connection-info';
import hotlists, { HotListState } from './hotlists';
import lines, { LineState } from './lines';
import nicklists, { NicklistState } from './nicklists';

type AppState = {
  connected: boolean;
  currentBufferId: string | null;
};

export type StoreState = {
  app: AppState;
  connection: ConnectionInfo;
  buffers: BufferState;
  lines: LineState;
  hotlists: HotListState;
  nicklists: NicklistState;
};

const initialState: AppState = {
  connected: false,
  currentBufferId: null
};

const app = (
  state: AppState = initialState,
  action: { type: string; bufferId?: string }
) => {
  switch (action.type) {
    case 'DISCONNECT':
      return {
        ...state,
        connected: false
      };
    case 'FETCH_VERSION':
      return {
        ...state,
        connected: true
      };
    case 'CHANGE_CURRENT_BUFFER':
      return {
        ...state,
        currentBufferId: action.bufferId
      };
    default:
      return state;
  }
};

export const reducer = combineReducers({
  app,
  buffers,
  lines,
  connection,
  hotlists,
  nicklists
});

export const store = configureStore({
  reducer: persistReducer(
    { storage: AsyncStorage, key: 'state', whitelist: ['connection'] },
    reducer
  ),
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [
          // https://github.com/rt2zz/redux-persist/issues/988
          FLUSH,
          REHYDRATE,
          PAUSE,
          PERSIST,
          PURGE,
          REGISTER,
          // The following actions send the raw relay payload to the reducer,
          // which may contain non-serializable data.
          // FIXME: actions should be serializable.
          'FETCH_HOTLISTS',
          'FETCH_LINES',
          'BUFFER_LINE_ADDED'
        ],
        // The following state entries contain the raw relay payload which may
        // contain non-serializable data. The data is not persisted.
        // FIXME: state should be serializable.
        ignoredPaths: [/^lines\.*'/, /^hotlists\./, /^lines\./]
      }
    })
});

export const persistor = persistStore(store);
