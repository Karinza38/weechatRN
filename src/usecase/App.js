import React from 'react-native';
import { connect } from 'react-redux/native';

import SideMenu from 'react-native-side-menu';

let {
    Component,
    View,
    Text,
    StyleSheet,
    } = React;

import { changeCurrentBuffer } from './buffers/actions/BufferActions';

import BufferView from './buffers/ui/BufferView';
import BufferList from './buffers/ui/BufferList';

class App extends Component {

    changeCurrentBuffer(bufferName) {
        this.props.dispatch(changeCurrentBuffer(bufferName));
    }
    render() {
        let { buffers, currentBufferName } = this.props;

        return (
            <View style={styles.container}>
                <View style={styles.topbar} />
                <SideMenu touchToClose={true} menu={<BufferList buffers={buffers}
                        currentBufferName={currentBufferName}
                        onSelectBuffer={(b) => this.changeCurrentBuffer(b.name)} />}>
                    <BufferView bufferName={currentBufferName}/>
                </SideMenu>
            </View>
        );
    }
}

export default connect(state => {
    return {
        currentBufferName: state.buffer.currentBufferName,
        buffers: state.buffer.buffers
    };
})(App);

const styles = StyleSheet.create({
    topbar: {
        height: 20,
        backgroundColor: '#001'
    },
    container: {
        flex: 1,
        backgroundColor: '#89a'
    }
});