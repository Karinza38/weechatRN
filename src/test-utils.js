// test-utils.js
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { reducer } from '../src/store';

function render(
  ui,
  {
    preloadedState = {},
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
