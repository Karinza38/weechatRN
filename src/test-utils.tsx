import { Store, configureStore } from '@reduxjs/toolkit';
import {
  RenderOptions,
  render as rtlRender
} from '@testing-library/react-native';
import React, { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';

import { StoreState, reducer } from './store';

interface ExtendedRenderOptions extends RenderOptions {
  preloadedState?: Partial<StoreState>;
  store?: Store<StoreState>;
}

function render(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren): React.JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
