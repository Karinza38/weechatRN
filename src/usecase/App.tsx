import * as React from "react";
import {
  SafeAreaView,
  View,
  Text,
  Image,
  TouchableOpacity,
  StyleSheet,
  Keyboard
} from "react-native";
import { connect } from "react-redux";
import * as _ from "lodash";

import Drawer from "react-native-drawer";

import { changeCurrentBuffer } from "./buffers/actions/BufferActions";

import BufferContainer from "./buffers/ui/BufferContainer";
import BufferList from "./buffers/ui/BufferList";
import { StoreState } from "../store";
import { renderWeechatFormat } from "../lib/weechat/color-formatter";

interface Props {
  buffers: { [key: string]: WeechatBuffer };
  currentBufferId: string | null;
  currentBuffer: WeechatBuffer | null;
  hasHighlights: boolean;
  disconnect: () => void;
  fetchLinesForBuffer: (bufferId: string) => void;
  sendMessageToBuffer: (fullBufferName: string, message: string) => void;
  clearHotlistForBuffer: (fullBufferName: string) => void;
  dispatch: (any) => void;
}

interface State {
  showTopic: boolean;
}

class App extends React.Component<Props, State> {
  drawer: Drawer;

  state: State = {
    showTopic: false
  };

  changeCurrentBuffer = (buffer: WeechatBuffer) => {
    const { currentBufferId, fetchLinesForBuffer } = this.props;

    this.drawer.close();
    if (currentBufferId !== buffer.id) {
      this.props.dispatch({
        type: "CHANGE_CURRENT_BUFFER",
        bufferId: buffer.id
      });
      this.props.clearHotlistForBuffer(buffer.full_name);
      fetchLinesForBuffer(buffer.id);
    }
  };

  toggleShowTopic = () => {
    this.setState(state => ({
      showTopic: !state.showTopic
    }));
  };

  openDrawer = () => {
    this.drawer.open();
    Keyboard.dismiss();
  };

  sendMessage = (message: string) => {
    const { currentBuffer, sendMessageToBuffer } = this.props;

    sendMessageToBuffer(currentBuffer.full_name, message);
  };

  render() {
    const {
      buffers,
      currentBufferId,
      currentBuffer,
      fetchLinesForBuffer,
      hasHighlights
    } = this.props;

    const { showTopic } = this.state;

    const sidebar = (
      <BufferList
        buffers={_.orderBy(_.values(buffers), ["number"])}
        currentBufferId={currentBufferId}
        onSelectBuffer={this.changeCurrentBuffer}
      />
    );

    return (
      <View style={styles.container}>
        <Drawer
          type="static"
          content={sidebar}
          panOpenMask={0.03}
          tapToClose={true}
          openDrawerOffset={100}
          captureGestures={true}
          ref={d => (this.drawer = d)}
          tweenHandler={Drawer.tweenPresets.parallax}
        >
          <SafeAreaView style={styles.container}>
            <View style={styles.topbar}>
              <View style={styles.channels}>
                <TouchableOpacity
                  style={styles.channelsButton}
                  onPress={this.openDrawer}
                >
                  <Text
                    style={[
                      styles.channelsButtonText,
                      hasHighlights && {
                        color: "#ffcf7f"
                      }
                    ]}
                  >
                    #
                  </Text>
                </TouchableOpacity>
              </View>
              <TouchableOpacity onPress={this.toggleShowTopic}>
                <Text style={styles.topbarText}>
                  {currentBuffer && currentBuffer.short_name}
                </Text>
              </TouchableOpacity>
              <View style={styles.channels}>
                <TouchableOpacity
                  style={styles.channelsButton}
                  onPress={this.props.disconnect}
                >
                  <Image source={require("./icons/eject.png")} />
                </TouchableOpacity>
              </View>
            </View>
            <BufferContainer
              showTopic={showTopic}
              buffer={currentBuffer}
              sendMessage={this.sendMessage}
              fetchLinesForBuffer={fetchLinesForBuffer}
              bufferId={currentBufferId}
            />
          </SafeAreaView>
        </Drawer>
      </View>
    );
  }
}

export default connect((state: StoreState) => {
  const currentBufferId = state.app.currentBufferId;
  const currentBuffer = currentBufferId && state.buffers[currentBufferId];

  const numHighlights = _.values(state.hotlists).reduce(
    (sum, hlist) => sum + hlist.highlight,
    0
  );

  return {
    buffers: state.buffers,
    currentBufferId,
    currentBuffer,
    hasHighlights: numHighlights > 0
  };
})(App);

const styles = StyleSheet.create({
  topbar: {
    flexDirection: "row",
    backgroundColor: "#333",
    justifyContent: "space-between",
    alignItems: "center",
    paddingBottom: 10
  },
  channels: {
    paddingHorizontal: 5
  },
  channelsButton: {
    paddingVertical: 5,
    paddingHorizontal: 10,
    width: 40
  },
  channelsButtonText: {
    textAlign: "center",
    fontSize: 20,
    fontFamily: "Gill Sans",
    color: "#eee",
    fontWeight: "bold"
  },
  topbarText: {
    color: "#eee",
    fontFamily: "Thonburi",
    fontWeight: "bold",
    fontSize: 15
  },
  container: {
    flex: 1,
    backgroundColor: "#333"
  }
});
