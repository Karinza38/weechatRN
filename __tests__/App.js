import 'react-native';
import React from 'react';
import App from '../src/usecase/App';

import { render } from '../src/test-utils';

it('renders correctly', () => {
  jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

  const tree = render(<App />, {
    initialState: {
      buffers: {
        1: {
          id: '1',
          full_name: 'irc.freenode.#weechat',
          short_name: '#weechat',
          local_variables: { type: 'channel' }
        }
      }
    }
  });

  expect(tree).toMatchSnapshot();
});
