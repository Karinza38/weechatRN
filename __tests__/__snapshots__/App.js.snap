// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#333",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      animatedStyle={
        {
          "value": {},
        }
      }
      collapsable={false}
      forwardedRef={[Function]}
      handlerTag={1}
      handlerType="PanGestureHandler"
      onGestureHandlerEvent={[Function]}
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        animatedStyle={
          {
            "value": {
              "transform": [
                {
                  "translateX": 320,
                },
              ],
            },
          }
        }
        collapsable={false}
        style={
          {
            "flex": 1,
            "transform": [
              {
                "translateX": 320,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={
            {
              "flex": 1,
            }
          }
        >
          <RNCSafeAreaView
            edges={
              {
                "bottom": "additive",
                "left": "additive",
                "right": "additive",
                "top": "additive",
              }
            }
            style={
              {
                "backgroundColor": "#333",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#333",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 10,
                }
              }
            >
              <View
                style={
                  {
                    "paddingHorizontal": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                      "width": 40,
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#eee",
                          "fontFamily": "Gill Sans",
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        },
                        false,
                      ]
                    }
                  >
                    #
                  </Text>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#eee",
                      "fontFamily": "Thonburi",
                      "fontSize": 15,
                      "fontWeight": "bold",
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "paddingHorizontal": 5,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                      "width": 40,
                    }
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "backgroundColor": "#222",
                  "flex": 1,
                }
              }
            />
          </RNCSafeAreaView>
        </View>
        <View
          accessibilityElementsHidden={false}
          animatedStyle={
            {
              "value": {
                "opacity": 1,
                "zIndex": 0,
              },
            }
          }
          collapsable={false}
          importantForAccessibility="auto"
          pointerEvents="auto"
          style={
            {
              "backgroundColor": "rgba(0, 0, 0, 0.5)",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            accessibilityLabel="Close drawer"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "flex": 1,
                "pointerEvents": "auto",
              }
            }
          />
        </View>
      </View>
      <View
        animatedStyle={
          {
            "value": {
              "transform": [
                {
                  "translateX": 0,
                },
              ],
            },
          }
        }
        collapsable={false}
        removeClippedSubviews={false}
        style={
          {
            "backgroundColor": "#121212",
            "bottom": 0,
            "maxWidth": "100%",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingTop": 0,
            "position": "absolute",
            "top": 0,
            "transform": [
              {
                "translateX": 0,
              },
            ],
            "width": 320,
            "zIndex": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#121212",
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            data={
              [
                {
                  "full_name": "irc.freenode.#weechat",
                  "id": "1",
                  "local_variables": {
                    "type": "channel",
                  },
                  "short_name": "#weechat",
                },
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={0.0001}
            stickyHeaderIndices={[]}
            style={
              {
                "backgroundColor": "#121212",
                "flex": 1,
              }
            }
            viewabilityConfigCallbackPairs={[]}
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "flex": 1,
                        "height": 40,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#eee",
                              "fontFamily": "Thonburi",
                              "fontSize": 15,
                              "fontWeight": "bold",
                            },
                            null,
                          ]
                        }
                      >
                        #weechat
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
